modules/bench/src/main/clojure/xtdb/bench2/xtdb2.clj

```diff
@@ -192,6 +194,49 @@
         worker (b/->Worker node root-random domain-state custom-state clock reports)]
     worker))
 
+(defn- only-oltp-stage [report]
+  (let [stage-filter #(filter (comp #{:oltp} :stage) %)]
+    (-> report
+        (update :stages stage-filter)
+        (update :metrics stage-filter))))
+
+(defn run-auctionmark [{:keys [output-file node-dir load-phase duration]
+                        :or {node-dir "dev/auctionmark-run"
+                             duration "PT30S"} :as opts}]
+  (let [output-file (or output-file (str "auctionmark-" duration ".edn"))
+        node-dir (.toPath (io/file node-dir))]
+    (when load-phase
+      (util/delete-dir node-dir))
+    (let [report (-> (run-benchmark
+                      {:node-opts {:node-dir node-dir
+                                   :instant-src InstantSource/SYSTEM}
+                       :benchmark-type :auctionmark
+                       :benchmark-opts (assoc opts :sync true)})
+                     only-oltp-stage)]
+      (spit (io/file output-file) report))))
+
+(def cli-options
+  [[nil "--load-phase LOAD-PHASE" :parse-fn #(if (or (= % "false") (= % "nil")) false true)]
+   [nil "--output-file OUTPUT-FILE"]
+   [nil "--node-dir NODE-DIR"]
+   [nil "--duration DURATION" :validate [(fn [x] (try (Duration/parse x)
```

Comment:
I'd have guessed that throwing an error from `:parse-fn #(Duration/parse %)` would do the same thing?